examples/todomvc-todo-manager/src/__tests__/index.test.ts

```diff
@@ -0,0 +1,286 @@
+import {
+  FeatureServiceBinder,
+  FeatureServiceConsumerEnvironment
+} from '@feature-hub/core';
+import {TodoManagerState, TodoManagerV1, todoManagerDefinition} from '..';
+
+describe('todoManagerDefinition', () => {
+  let mockEnv: FeatureServiceConsumerEnvironment;
+  let spyConsoleInfo: jest.SpyInstance;
+
+  beforeEach(() => {
+    mockEnv = {featureServices: {}, config: {}};
+
+    spyConsoleInfo = jest.spyOn(console, 'info');
+    spyConsoleInfo.mockImplementation(jest.fn());
+  });
+
+  afterEach(() => {
+    spyConsoleInfo.mockRestore();
+  });
+
+  it('defines a todo manager', () => {
+    expect(todoManagerDefinition.id).toBe('s2:todomvc-todo-manager');
+    expect(todoManagerDefinition.dependencies).toBeUndefined();
+  });
+
+  describe('#create', () => {
+    it('creates a shared todo manager containing version 1.0', () => {
+      const sharedTodoManager = todoManagerDefinition.create(mockEnv);
+
+      expect(sharedTodoManager['1.0']).toBeDefined();
+    });
+  });
+
+  describe('TodoManagerV1', () => {
+    let todoManagerV1: TodoManagerV1;
+
+    beforeEach(() => {
+      const sharedTodoManager = todoManagerDefinition.create(mockEnv);
+
+      todoManagerV1 = (sharedTodoManager['1.0'] as FeatureServiceBinder<
+        TodoManagerV1
+      >)('test:consumer').featureService;
+    });
+
+    describe('#add', () => {
+      describe('with no initial todos', () => {
+        it('adds the todo', () => {
+          todoManagerV1.add('test');
+
+          const expectedTodos: TodoManagerState = {
+            allIds: ['0'],
+            byId: {
+              0: {id: '0', title: 'test', completed: false}
+            }
+          };
+
+          expect(todoManagerV1.getTodos()).toEqual(expectedTodos);
+        });
+      });
+
+      describe('with multiple todos', () => {
+        it('adds the todos', () => {
+          todoManagerV1.add('test 1');
+          todoManagerV1.add('test 2');
+          todoManagerV1.add('test 3');
+
+          const expectedTodos: TodoManagerState = {
+            byId: {
+              0: {id: '0', title: 'test 1', completed: false},
+              1: {id: '1', title: 'test 2', completed: false},
+              2: {id: '2', title: 'test 3', completed: false}
+            },
+            allIds: ['0', '1', '2']
+          };
+
+          expect(todoManagerV1.getTodos()).toEqual(expectedTodos);
+        });
+      });
+
+      it('logs the added todo', () => {
+        todoManagerV1.add('test');
```

Comment:
Just a note for a later refactoring: The add method should return the created Todo or its ID. Otherwise you can not create a Todo programmatically and set it to completed afterwards, which is something I wanted to do in the past.